### ISSUE TITLE
[java] False negatives in AssignmentInOperand Rule

### ISSUE THREAD

**Issue by Noah0120** [pmd/pmd#3434]
**Affects PMD Version:** 
6.36.0+
7.0.0

**Rule:** [AssignmentInOperand](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#assignmentinoperand)

**Description:**
The term `Operand` means that `subexpression of. another expression` based on #3429.
Therefore, assignment in another expression should be detected.
However, PMD didn't detect the following five cases:
1) `assignment` in a `method argument`
2) `assignment` in a `conditional expression`
3) `assignment` in an `array index`
4) `assignment` in a `method caller`
5) `assignment` in a `constructor` 

**Code Sample demonstrating the issue:**

```java
class AssignmentInOperand{
        void test(){
        // case1
        keys.contains(k = random.nextLong());
        Character.isJavaIdentifierPart(text.charAt(--j));

        // case2
        addNextStaticMember = ((staticMember.getNextStaticMember() != null) ? true : false) ;
        ((null == theClassMethods) ? (theClassMethods = theClass.getMethods()) : theClassMethods); 

        // case3
        Character.isWhitespace(segment.array[offset++]);
        subPropMatches[i++];

        // case4
        (is = ia.getNext(null)).isPresent();
        (destFile = new File (dest, srcFO.getNameExt ())).exists();

        // case5
        new File(location, projectName = projectNamePrefix + "_" + num).exists();
        }
}
```

**Expected outcome:**
These are assignments in another expression.
Therefore, it should be detected.
These are false negatives.

Thank you.

**Running PMD through:** *[CLI]*

**Comment by VoidxHoshi** [comment]
Well, let me have a try.
